spec/services/partner_profile_update_service_spec.rb

```diff
@@ -81,6 +82,18 @@
           expect(profile.served_areas.size).to eq(2)
         end
       end
+
+      context "and the new values include country but are missing client share" do
+        it "maintains the old values and returns the correct validation error" do
+          profile.reload
+          expect(profile.served_areas.size).to eq(2)
+          result = PartnerProfileUpdateService.new(profile.partner, partner_params, incorrect_attributes_missing_client_share).call
+          expect(result.success?).to eq(false)
+          expect(result.error.to_s).to include("Validation failed: Served areas client share is not a number, Served areas client share is not included in the list")
```

Comment:
It is a somewhat confusing error.   I wrote the code, and I'm not even quite sure what "Served areas client share is not included in the list" means  when the county I picked is in the list of counties.     However,  it's probably out of scope for this change.